realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import io.realm.annotations.Beta;
+import io.realm.internal.ManagableObject;
+import io.realm.internal.Row;
+
+
+/**
+ * A MutableRealmInteger is a mutable, {@link java.lang.Long}-like numeric quantity.
+ * MutableRealmInteger implementations are not thread safe.  Like all Realm objects,
+ * managed RealmIntegers may not be moved across threads.  Unmanaged RealmObject
+ * may be moved across threads but require safe publication.
+ * <p>
+ * It wraps an internal CRDT counter:
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type"></a>
+ * <p>
+ * TODO: More complete docs, including examples of use.
+ */
+@Beta
+public abstract class MutableRealmInteger implements Comparable<MutableRealmInteger>, ManagableObject {
+
+    /**
+     * Unmanaged Implementation.
+     */
+    private static final class Unmanaged extends MutableRealmInteger {
+        private Long value;
+
+        Unmanaged(Long value) {
+            this.value = value;
+        }
+
+        @Override
+        public void set(Long newValue) {
+            value = newValue;
+        }
+
+        @Override
+        public Long get() {
+            return value;
+        }
+
+        @Override
+        public void increment(long inc) {
+            if (value == null) {
+                throw new NullPointerException("Attempt to increment a null valued MutableRealmInteger");
+            }
+            value = Long.valueOf(value + inc);
+        }
+
+        @Override
+        public void decrement(long dec) {
+            if (value == null) {
+                throw new NullPointerException("Attempt to decrement a null valued MutableRealmInteger");
+            }
+            value = Long.valueOf(value - dec);
+        }
+
+        @Override
+        public boolean isManaged() {
+            return false;
+        }
+
+        @Override
+        public boolean isValid() {
+            return true;
+        }
+    }
+
+
+    /**
+     * Managed Implementation.
+     */
+    // FIXME Counters: wire up to native increment method
+    static class Managed extends MutableRealmInteger {
+        private final ProxyState<?> proxyState;
+        private final BaseRealm realm;
+        private final Row row;
+        private final long columnIndex;
+        private Long value;
```

Comment:
Yes, surely.  This is an intermediate commit.  I needed some way to make existing tests work.  Value gets removed.  Added FIXME comment